protocols/kad/src/query/peers/closest.rs

```diff
@@ -691,4 +693,78 @@ mod tests {
 
         QuickCheck::new().tests(10).quickcheck(prop as fn(_) -> _)
     }
+
+    #[test]
+    fn stalled_iter_at_capacity_use_max_of_parallelism_and_num_results() {
```

Comment:
Wouldn't this make for a good property test? The property being the following: "For any `ClosestPeersIter`, if it is in `State::Stalled` then it must be at capacity only if `num_waiting = max{num_results, parallelism}`" ?